### PR TITLE
Update tests workflow with Codecov version 3

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**

Update Codecov version to 3 in the tests workflow.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -79,7 +79,7 @@ jobs:
         run: bash <(sed -i 's/filename=\"/filename=\"pennylane\//g' coverage.xml)
 
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v1
+        uses: codecov/codecov-action@v2
```
Comment (rmoyard) [inline .github/workflows/tests.yml:82]:
```suggestion
        uses: codecov/codecov-action@v3
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -146,7 +146,7 @@ jobs:
...
```
Comment (rmoyard) [inline .github/workflows/tests.yml:149]:
```suggestion
        uses: codecov/codecov-action@v3
```

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -191,11 +191,11 @@ jobs:
           cd qchem && python -m pytest tests --cov=pennylane_qchem $COVERAGE_FLAGS
 
       - name: Upload HF coverage to Codecov
-        uses: codecov/codecov-action@v1
+        uses: codecov/codecov-action@v2
```
Comment (rmoyard) [inline .github/workflows/tests.yml:194]:
```suggestion
        uses: codecov/codecov-action@v3
```

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -191,11 +191,11 @@ jobs:
...
 
       - name: Upload HF coverage to Codecov
-        uses: codecov/codecov-action@v1
+        uses: codecov/codecov-action@v2
         with:
           file: ./coverage.xml
 
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v1
+        uses: codecov/codecov-action@v2
```
Comment (rmoyard) [inline .github/workflows/tests.yml:199]:
```suggestion
        uses: codecov/codecov-action@v3
```

**Review by antalszava — APPROVED** [review]
:tada:

**Pull request merged**